pkgs/servers/home-assistant/custom-lovelace-modules/xiaomi-vacuum-map-card/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  runCommand,
+  jq,
+  buildNpmPackage,
+  fetchFromGitHub,
+}:
+
+buildNpmPackage rec {
+  pname = "xiaomi-vacuum-map-card";
+  version = "2.3.1";
+
+  src = fetchFromGitHub {
+    owner = "PiotrMachowski";
+    repo = "lovelace-xiaomi-vacuum-map-card";
+    rev = "v${version}";
+    hash = "sha256-VBcTcj7SNULusChDAGqheCPkEmpqWDaBt2blCvBVkCc=";
+  };
+
+  # The version used upstream is oudated and does not work with our newer nodejs versions.
+  patchedPackageJSON = runCommand "package.json" { } ''
+    ${lib.getExe jq} '.devDependencies."rollup-plugin-typescript2" = "^0.36.0"' ${src}/package.json > $out
+  '';
+
+  postPatch = ''
+    cp ${patchedPackageJSON} ./package.json
+    cp ${./package-lock.json} ./package-lock.json
+  '';
```

Comment:
Did you file an issue so we can eventually get rid of this workaround?

Reply:
[yes](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/issues/851)